webrtc/RTCPeerConnection-setRemoteDescription.html

```diff
@@ -44,6 +46,42 @@
       assert_unreached('Error ' + e.name + ': ' + e.message);
     }));
   }, 'Triggers ontrack when called with a remote description and the MSID of the stream is is parsed.');
+
+  promise_test(t => {
+    const pc = new RTCPeerConnection();
+    return promise_rejects(t, new RTCError(),
+      pc.setRemoteDescription({
+        // valid SDP type
+        type: 'offer',
+        // malformed SDP description
+        sdp: 'bogus'
+      }));
+  }, 'Malformed SDP description should be rejected with RTCError');
+
+  promise_test(t => {
+    const pc = new RTCPeerConnection();
+    return promise_rejects(t, new TypeError(),
+      pc.setRemoteDescription({
+        // invalid enum value is caught at IDL level before
+        // method is executed
+        type: 'bogus',
+        // bogus SDP should never be validated before type
+        sdp: 'bogus'
+      }));
+  }, 'Invalid SDP type should be rejected with TypeError');
+
+  promise_test(t => {
+    const pc = new RTCPeerConnection();
+
+    return promise_rejects(t, new InvalidStateError(),
```

Comment:
I see. Didn't know that `InvalidStateError` is a `DOMException` instead of a class of its own. (webrtc-pc is missing quite a lot of links on references to the other specs)

Since this PR has been merged I will try to change it together with new tests I will be writing to `setRemoteDescription`.